lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1527,14 +1523,6 @@ private function validateAndCompleteTypedAssociationMapping(array $mapping): arr
             $mapping['targetEntity'] = $type->getName();
         }
 
-        if (isset($mapping['joinColumns'])) {
-            foreach ($mapping['joinColumns'] as &$joinColumn) {
-                if ($type->allowsNull() === false) {
```

Comment:
instead of removing it, do we wan the same logic as for `$mapping['nullable']`?
e.g. `if (! isset($joinColumn['nullable'])) $joinColumn['nullable'] = $type->allowsNull();`

Reply:
No, tthe column nullable behavior is also flawed, as there are legit cases where you want nullable behavior to be different between entity and database.

Reply:
ah, at least for me the bug was only that I could not set the value at all in the JoinColumn annotation. it was always false even when providing it explicitly.

but alright, I see setting the default based on the php type may be a bc where it is not defined in the annotation. so maybe this is really something for orm 3?